docs/source/developer/ui_components.rst

```diff
@@ -0,0 +1,337 @@
+
+Sync icon color to JupyterLab theme
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. raw:: html
+
+   <em>Example svgs with class annotation can be found in <a href="https://github.com/jupyterlab/jupyterlab/tree/f0153e0258b32674c9aec106383ddf7b618cebab/packages/ui-components/style/icons">ui-components/style/icons</a></em>
+
+| 
+| You can ensure that the colors of your custom ``LabIcon`` sync up to the colors of the current JuptyerLab theme by adding appropriate ``class`` annotations to each colored element of your icon's svg.
+| 
+| In other words, each element of your svg that a ``fill="..."`` or a ``stroke="..."`` property should also have a ``class="jp-icon<whatever>"`` property.
+
+Available icon classes
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <em>Icon-related CSS classes are defined in <a href="https://github.com/jupyterlab/jupyterlab/blob/f0153e0258b32674c9aec106383ddf7b618cebab/packages/ui-components/style/icons.css">ui-components/style/icons.css</a></em>
+
+| 
+| All colors shown are for the standard light/dark theme, mouse over for hex values.
+
+``jp-iconX``: contrast to theme background
+''''''''''''''''''''''''''''''''''''''''''
+
+.. raw:: html
+
+   <ul>
+   <li>jp-icon0: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#111"/><title>#111</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#fff"/><title>#fff</title></svg></li>
+   <li>jp-icon1: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#212121"/><title>#212121</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#fff"/><title>#fff</title></svg></li>
+   <li>jp-icon2: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#424242"/><title>#424242</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#eee"/><title>#eee</title></svg></li>
+   <li>jp-icon3: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#616161"/><title>#616161</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#bdbdbd"/><title>#bdbdbd</title></svg></li>
+   <li>jp-icon4: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#757575"/><title>#757575</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#757575"/><title>#757575</title></svg></li>
+   </ul>
+
+Most one-color icons in JupyterLab (including the sidebar and toolbar
+icons) are colored using the ``jp-icon3`` class.
+
+For light/dark themes, ``jp-icon0`` corresponds to the darkest/lighest
+background color, while ``jp-icon1`` is somewhat lighter/darker, and so
+forth.
+
+``jp-icon-accentX``: match to theme background
+''''''''''''''''''''''''''''''''''''''''''''''
+
+.. raw:: html
+
+   <ul>
+   <li>jp-icon-accent0: <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#fff"/><title>#fff</title></svg> / <svg width="16" viewBox="0 0 1 1"><rect width="1" height="1" fill="#111"/><title>#111</title></svg></li>
```

Comment:
Not sure if we should hard code the color values here. Maybe instead say semantically what the accents are for? Or link to where they are defined? Just imagining if we ever change these values we might forget to update the docs?

Reply:
This is supposed to render as a list of color swatches. It shows up correctly in the built dev docs/my local markdown preview:

![image](https://user-images.githubusercontent.com/2263641/80433012-c1d23300-88c3-11ea-89e3-f89d66b51f1b.png)

but it doesn't look like it's showing up correctly on github (the inline svgs are getting scrubbed). What's it look like to you?

Reply:
I'm of the opinion that a picture is worth a thousand semantic explanations, especially when describing anything visual/having to do with design. So I think the potential tradeoffs are worth it... assuming the swatches render correctly.

These particular colors haven't changed in the last couple of years, and aren't likely to change significantly/frequently in the future. Longer term, we can look into auto-generating the documentation color swatches from the theme CSS.

Reply:
> but it doesn't look like it's showing up correctly on github (the inline svgs are getting scrubbed). What's it look like to you?

Yeah it didn't look like colors to me.

> These particular colors haven't changed in the last couple of years, and aren't likely to change significantly/frequently in the future. Longer term, we can look into auto-generating the documentation color swatches from the theme CSS.

OK sounds good to me! I am convinced.